### DISCUSSION TITLE
How do I get the GitHub CoPilot extension

### DISCUSSION THREAD

**Discussion by simonjcarr** [coder/code-server#6370] [General]
Is the GitHub CoPilot extension available for coder? If not, do we know if there are any plans for it to be available?

I can see why GitHub may not want to offer it, as coder is a competitor to there CodeSpaces offering.

I have tried to use the ChatGPT extension, but after installing and clicking the icon for it in the tool bar nothing happens other than the blue line traveling across the top as though it is trying to load but failing.

**Comment by code-asher** [comment]
code-server uses Open VSX as the default marketplace so making it 
available is a matter of the extension authors submitting it to 
Open VSX and I think they also have code that explicitly disables 
itself on non-Microsoft-branded builds of VS Code so they would 
need to remove that.

I see a few open discussions for this but no responses from 
Microsoft yet: 
https://github.com/orgs/community/discussions?discussions_q=is%3Aopen+open+vsx

As for the ChatGPT extension, is it the genieai.chatgpt-vscode 
extension?  If so that one worked last I tried 
(https://github.com/coder/code-server/issues/6117) and the issue 
for others was not serving code-server over a secure domain.

If that does not help then feel free to post logs and we can dig 
into it more.

**Reply by code-asher** [reply]
I should mention there are discussions here about how to work around installing Copilot but these methods probably violate Microsoft's marketplace terms of service and maybe the extension's license as well.
